### PR TITLE
sql: fix crdb_internal.zones(target) to show user-defined schemas

### LINKED ISSUE #61728: sql: crdb_internal.zones incorrectly assumes all tables are in the public schema
**Describe the problem**

The zone configs in crdb_internal.zones do not distinguish between tables in different schemas. It assumes everything is in the `public` schema.

**To Reproduce**
```
root@:26257/defaultdb> create schema s;
CREATE SCHEMA

root@:26257/defaultdb> create table s.t(a int primary key);
CREATE TABLE

root@:26257/defaultdb> ALTER TABLE s.t CONFIGURE ZONE USING range_min_bytes = 0, range_max_bytes = 90000, gc.ttlseconds = 89999, num_replicas = 1;
CONFIGURE ZONE 1

root@:26257/defaultdb> select target, database_name, table_name, raw_config_yaml from crdb_internal.zones where table_name = 't';;
           target          | database_name | table_name |    raw_config_yaml
---------------------------+---------------+------------+-------------------------
  TABLE defaultdb.public.t | defaultdb     | t          | range_min_bytes: 0
                           |               |            | range_max_bytes: 90000
                           |               |            | gc:
                           |               |            |   ttlseconds: 89999
                           |               |            | global_reads: null
                           |               |            | num_replicas: 1
                           |               |            | num_voters: null
                           |               |            | constraints: []
                           |               |            | voter_constraints: []
                           |               |            | lease_preferences: []
```

**Expected behavior**

Add a schema_name column, and make sure the existing target column shows the correct schema.

### PR DESCRIPTION
**Pull request by rafiss**
fixes #61728 

This also includes a 2nd commit that adds a schema_name column to
crdb_internal.zones.

Release note (bug fix): Previously the `target` column of
crdb_internal.zones would show names without properly accounting for
user-defined schemas. Now it does.

### CODE REVIEW

File: pkg/config/zonepb/zone_test.go [hunk 1]
```diff
@@ -984,8 +984,9 @@ func TestZoneSpecifiers(t *testing.T) {
 	}
 
 	// Simulate the following schema:
-	//   CREATE DATABASE db;   CREATE TABLE db.tbl ...
-	//   CREATE DATABASE carl; CREATE TABLE carl.toys ...
+	//   CREATE DATABASE db;        CREATE TABLE db.public.tbl ...
+	//   CREATE DATABASE carl;      CREATE TABLE carl.public.toys ...
+	//   CREATE SCHEMA test_schema; CREATE TABLE carl.test_schema.toys ...
```
Comment (otan) [inline pkg/config/zonepb/zone_test.go:989]: db.test_schema.toys?

File: pkg/sql/logictest/testdata/logic_test/crdb_internal [hunk 2]
```diff
@@ -277,11 +277,15 @@ SELECT * FROM crdb_internal.ranges_no_leases WHERE range_id < 0
 ----
 range_id  start_key  start_pretty  end_key  end_pretty  table_id  database_name  schema_name  table_name  index_name  replicas  replica_localities  learner_replicas  split_enforced_until
 
+statement ok
+CREATE SCHEMA schem; CREATE TABLE schem.bar (y INT PRIMARY KEY)
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/crdb_internal:281]: nit: i'm curious, whilst you're here, for completeness can you name the schema `"schema"`?

File: pkg/config/zonepb/zone_test.go (hunk 1 above)
Comment (otan) [reply pkg/config/zonepb/zone_test.go:989]: oops.

**Review by otan — APPROVED** [review]

File: pkg/sql/logictest/testdata/logic_test/crdb_internal (hunk 2 above)
Comment (rafiss) [reply pkg/sql/logictest/testdata/logic_test/crdb_internal:281]: done

**Comment by rafiss** [comment]
bors r=otan

**Pull request merged**